Flickr/Flickr/ImageCollectionViewController.swift

```diff
@@ -10,17 +10,32 @@
 import UIKit
 import Alamofire
 
-class ImageCollectionViewController: UIViewController, UICollectionViewDelegate, UICollectionViewDataSource, UITextFieldDelegate {
-  
+class ImageCollectionViewController: UIViewController, UICollectionViewDelegate, UICollectionViewDataSource, UITextFieldDelegate, UICollectionViewDelegateFlowLayout {
+    
     fileprivate var imageData: [Photo] = [Photo]()
+    var isSearching = false
+    var sizes: [CGSize] = [CGSize]()
     
     override func viewDidLoad() {
         super.viewDidLoad()
+        searchCollectionView.alpha = 1
+        popularCollectionView.alpha = 0
+        searchCollectionView.delegate = self
+        searchCollectionView.dataSource = self
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        Alamofire.request(Router.popular(page: 1)).responseJSON { (response) in
```

Comment:
after you get back on this screen viewWillAppear will be called again and make unexpected behaviour. Move this to viewDidLoad